// angular2-infinite-scroll/src/axis-resolver.js
"use strict";
var core_1 = require("@angular/core");
var AxisResolver = (function () {
    function AxisResolver() {
        this.setVertical(true);
    }
    AxisResolver.prototype.setVertical = function (vertical) {
        if (vertical === void 0) { vertical = true; }
        this.vertical = vertical;
    };
    AxisResolver.prototype.clientHeightKey = function () { return this.vertical ? 'clientHeight' : 'clientWidth'; };
    AxisResolver.prototype.offsetHeightKey = function () { return this.vertical ? 'offsetHeight' : 'offsetWidth'; };
    AxisResolver.prototype.scrollHeightKey = function () { return this.vertical ? 'scrollHeight' : 'scrollWidth'; };
    AxisResolver.prototype.pageYOffsetKey = function () { return this.vertical ? 'pageYOffset' : 'pageXOffset'; };
    AxisResolver.prototype.offsetTopKey = function () { return this.vertical ? 'offsetTop' : 'offsetLeft'; };
    AxisResolver.prototype.scrollTopKey = function () { return this.vertical ? 'scrollTop' : 'scrollLeft'; };
    AxisResolver.prototype.topKey = function () { return this.vertical ? 'top' : 'left'; };
    AxisResolver.decorators = [
        { type: core_1.Injectable },
    ];
    /** @nocollapse */
    AxisResolver.ctorParameters = [];
    return AxisResolver;
}());
exports.AxisResolver = AxisResolver;
